webdriver/tests/support/helpers.py

```diff
@@ -60,24 +60,47 @@ def _restore_window_state(session):
             session.window.size = defaults.WINDOW_SIZE
 
     @ignore_exceptions
-    def _restore_windows(session):
+    async def _restore_windows(session):
         """Close superfluous windows opened by the test.
 
         It will not end the session implicitly by closing the last window.
+        If available, uses WebDriver BiDi session. Otherwise, uses the WebDriver Classic.
         """
-        current_window = session.window_handle
 
-        for window in _windows(session, exclude=[current_window]):
-            session.window_handle = window
-            if len(session.handles) > 1:
-                session.window.close()
+        if session.bidi_session:
+            command_result_future = await session.bidi_session.send_command("browsingContext.getTree", {"maxDepth": 1})
+            command_result = await command_result_future
```

Comment:
Instead of the double await, could you switch to `send_blocking_command` instead?

Reply:
done